test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go

```diff
@@ -0,0 +1,233 @@
+				if err != nil {
+					t.Fatalf("Could not get secret: %v", err)
+				}
+				lastUsed, ok := liveSecret.GetLabels()[serviceaccount.LastUsedLabelKey]
+				if !ok {
+					t.Fatalf("The secret %s should be labeled lastUsed time: %s", test.secretName, lastUsed)
+				} else {
+					t.Logf("The secret %s has been labeled with %s", test.secretName, lastUsed)
+				}
+			}
+
+			_, err = c.CoreV1().Secrets(myns).Get(context.TODO(), test.secretName, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					t.Fatalf("The secret %s should not be cleaned up, err: %v", test.secretName, err)
+				} else {
+					t.Fatalf("Failed to get secret %s, err: %v", test.secretName, err)
+				}
+			}
+
+			fakeClock.Step(cleanUpPeriod + 24*time.Hour)
+			time.Sleep(syncInterval)
+			liveSecret, err = c.CoreV1().Secrets(myns).Get(context.TODO(), test.secretName, metav1.GetOptions{})
+			if test.expectCleanedUp && err == nil {
+				t.Fatalf("The secret %s should be cleaned up. time: %v; creationTime: %v", test.secretName, fakeClock.Now().UTC(), liveSecret.CreationTimestamp)
+			} else if !test.expectCleanedUp && err != nil {
+				if apierrors.IsNotFound(err) {
+					t.Fatalf("The secret %s should not be cleaned up, err: %v", test.secretName, err)
+				} else {
+					t.Fatalf("Failed to get secret %s, err: %v", test.secretName, err)
+				}
+			}
+		})
+	}
+}
+
+func startLegacyServiceAccountTokenCleaner(ctx context.Context, client clientset.Interface, fakeClock clock.Clock, informers clientinformers.SharedInformerFactory) {
+	legacySATokenCleaner, _ := serviceaccountcontroller.NewLegacySATokenCleaner(
+		informers.Core().V1().ServiceAccounts(),
+		informers.Core().V1().Secrets(),
+		informers.Core().V1().Pods(),
+		client,
+		fakeClock,
+		serviceaccountcontroller.LegacySATokenCleanerOptions{
+			SyncInterval:  syncInterval,
+			CleanUpPeriod: cleanUpPeriod,
+		})
+	go legacySATokenCleaner.Run(ctx)
+	informers.Start(ctx.Done())
+}
+
+func createAutotokenMountedPod(c clientset.Interface, ns, secretName string) (*v1.Pod, error) {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "token-bound-pod",
+			Namespace: ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "name", Image: "image"},
+			},
+			Volumes: []v1.Volume{{Name: "foo", VolumeSource: v1.VolumeSource{Secret: &v1.SecretVolumeSource{SecretName: secretName}}}},
+		},
+	}
+	pod, err := c.CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create pod with token (%s:%s) bound, err: %v", ns, secretName, err)
+	}
+	err = wait.Poll(time.Second, 10*time.Second, func() (bool, error) {
+		pod, err = c.CoreV1().Pods(ns).Get(context.TODO(), "token-bound-pod", metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("failed to get pod with token (%s:%s) bound, err: %v", ns, secretName, err)
+		}
+		return true, nil
+	})
```

Comment:
this is polling the wrong thing... this is checking the API, which will always return the pod immediately... this just happens to deflake the test because it inserts a 1 second delay

I think you need to poll the pod informer in use to wait until the created pod is observed by the informer, and use PollImmediate to avoid the initial delay